sklearn/decomposition/tests/test_kernel_pca.py

```diff
@@ -523,3 +523,28 @@ def test_kernel_pca_feature_names_out():
 
     names = kpca.get_feature_names_out()
     assert_array_equal([f"kernelpca{i}" for i in range(2)], names)
+
+
+def test_kernel_pca_inverse_correct_gamma():
+    """Check that gamma is set correctly when not provided.
+
+    Non-regression test for #26280
+    """
+    rng = np.random.RandomState(0)
+    X = rng.random_sample((5, 4))
+
+    kpca1 = KernelPCA(
+        n_components=2, random_state=rng, fit_inverse_transform=True, kernel="rbf"
+    ).fit(X)
```

Comment:
Just a small nitpick, but I think it would be nice to set `gamma=None` explicitly just to make it clearer at a glance what is being compared. Additionally you could also set the shared arguments as a dictionary that you unpack in both, just to minimize clutter.